### PR TITLE
Replicate Magic Fool to clients

### PR DESCRIPTION
**Pull request by darshan3**
NPC is now visible at both server and client side.

### How to test:
1. Host a server.
2. Interact with NPC.
3. Join the server using a 2nd client.
4. Interact with NPC.

### CODE REVIEW

**Comment by darshan3** [comment]
Current Issue: When a client activates dialog, it shows up on both server as well as the client.

File: src/main/java/org/terasology/las/platform/GuidanceNpcProvider.java [hunk 1]
```diff
@@ -36,19 +37,16 @@
...
             EntityStore entity = new EntityStore(chooseFactionNpc);
             entity.addComponent(new LocationComponent(new Vector3f(0, 61, 24)));
-            entity.addComponent(dialogComp);
             buffer.enqueue(entity);
 
             Prefab beaconMark = Assets.getPrefab("BeaconMark").get();
             EntityStore entityStore = new EntityStore(beaconMark);
             entityStore.addComponent(new LocationComponent(new Vector3f(0, 63, 24)));
-
+            entityStore.addComponent(new NetworkComponent());
```
Comment (jellysnake) [inline src/main/java/org/terasology/las/platform/GuidanceNpcProvider.java]: Is there a reason for adding this component through code over prefab?

Comment (darshan3) [reply src/main/java/org/terasology/las/platform/GuidanceNpcProvider.java]: 'BeaconMark' prefab is taken from Quests module. Hence, adding network component would require changing that module. Should I create beaconmark prefab for this module too?

Comment (jellysnake) [reply src/main/java/org/terasology/las/platform/GuidanceNpcProvider.java]: I'd suggest looking into "delta"s. They allow you to modify a prefab from another module.
[See this guide for more info](https://github.com/Terasology/TutorialAssetSystem/wiki/Deltas-and-Overrides)

**Pull request merged**